### ISSUE TITLE
Heatmap and other ideas to improve bestof.js.org

### ISSUE THREAD

**Issue by michaelrambeau** [michaelrambeau/bestofjs-webui#33]
Some ideas to discuss, to make bestof.js.org better...

### A. New layout

The left sidebar is too long, there are too many tags, we could remove it and use the search field to select one or several tags, maybe adding an autocomplete feature
And then the home page could become a wall of project "cards", whose number of columns depends on the width of the screen.

### B. Project heatmap

When opening a project page, it would be nice to see the star trends over the last months using a heat map, similar to the heat map used in Github repo page to show user’s activity.

![image](https://cloud.githubusercontent.com/assets/5546996/21080049/4a5bc72a-bfe8-11e6-8d90-07886a93c5f5.png)

![image](https://cloud.githubusercontent.com/assets/5546996/21080085/3fddf9c0-bfe9-11e6-8912-7fe86b69b393.png)

### C. Tag trends graph

When opening a given tag page, it would be nice to see a graph to be able to compare trends over the last months.

**Comment by michaelrambeau** [comment]
## Heatmap implementation

### Frontend

The heatmap has been implemented in the following commit: https://github.com/michaelrambeau/bestofjs-webui/commit/fd6504a9beb791f9f6dfefd9526021bc48b37498

Inside the project page `/projects/:slug`, a new block called "DAILY TRENDS" displays the heatmap and then the trends for the project:

* Stars added yesterday
* Stars added over the last week
* Stars add over the last month
* Stars added over the last 3 months
* Stars added over the last 12 months

#### Screenshot (Preact project)

![image](https://user-images.githubusercontent.com/5546996/30529621-3dd87114-9c7a-11e7-8d53-a702b8fdfe04.png)

### Backend

#### `bestofjs-batches`

Everyday, the batch called `github` takes a snapshot of GitHub stargazers and saves inside the project document an array that contains the daily variations of the stargazers, for the last 365 days.

#### `bestofjs-api`

When calling the endpoint `/projects/:slug`, the API version 1 returns the array of the daily variations saved inside  `project` records in the database.

JSON response for https://bestofjs-api-v1.now.sh/projects/GoogleChrome/puppeteer
(`daily-trends` array contains only 10 values because the project has been added recently)

```json
{
  "name": "Lozad",
  "github": {
    "name": "lozad.js",
    "full_name": "ApoorvSaxena/lozad.js",
    "description": "🔥  Highly performant, light ~0.5kb and configurable lazy loader in pure JS with no dependencies for images, iframes and more",
    "homepage": "https://apoorv.pro/lozad.js/demo/",
    "stargazers_count": 1876,
    "pushed_at": "2017-09-15T18:58:18.000Z",
    "owner_id": 307583,
    "branch": "master",
    "topics": ["performance", "javascript", "lazy-loading", "lazyload", "lozad"]
  },
  "tags": ["load"],
  "npm": { "name": "lozad", "version": "1.0.2", "dependencies": [] },
  "icon": "",
  "daily-trends": [68, 125, 202, 230, 361, 331, 85, 22, 66]
}
```

**Comment by michaelrambeau** [comment]
The heatmap, made with https://github.com/patientslikeme/react-calendar-heatmap, has been included in [0.9.0 release](https://github.com/michaelrambeau/bestofjs-webui/releases/tag/0.9.0).
